src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -418,9 +426,39 @@ func (c controller) postInstallConfigs(ctx context.Context) error {
 		if err = c.updatePendingOLMOperators(); err != nil {
 			return errors.Errorf("Timeout while waiting for some of the operators and not able to update its state")
 		}
-		c.log.Warnf("Timeout while waiting for OLM operators be installed")
+		c.log.WithError(err).Warnf("Timeout while waiting for OLM operators be installed")
+		// Don't return the error here so the installation is marked as sucessful.
+		// We only log the error, and update the state of the monitored operators in assisted-service API.
+	}
+
+	return nil
+}
+
+func (c controller) applyPostInstallManifests() error {
+	ctx := utils.GenerateRequestContext()
+	tempDir, err := ioutil.TempDir("", "controller-custom-manifests-")
```

Comment:
not sure what can happened but will we retry on error?

Reply:
We retry only on `oc apply` failures, do you think we should retry in all the errors?

Reply:
yep

Reply:
We should retry in all errors but we should avoid repeating successful steps, meaning if you already downloaded the files there is no need to download them again.

Reply:
sounds good idea to delete file only on success and to check if it exists before download.
Though can be optimization in future PR

Reply:
I just wonder how often this could happen, so it should be optimalized?